src/Mvc/Mvc.Core/test/ApplicationModels/InferParameterBindingInfoConventionTest.cs

```diff
@@ -1094,6 +1115,14 @@ private class ParameterBindingNoRoutesOnController
 
     private class TestModel { }
 
+    private class CompositeTestModel
+    {
+        [FromQuery]
+        public int Id { get; set; }
+
+        public TestModel TestModel { get; set; }
```

Comment:
This `TestModel` is no longer bound after the change, right? And before it would have been bound from the body? Would the `[FromQuery]` previously overwrite the `Id` from the body with and id from the query string? If so, could people be relying on this?

Reply:
> This `TestModel` is no longer bound after the change, right?

It will be bound FromRoute or Query

Reply:
> And before it would have been bound from the body? Would the `[FromQuery]` previously overwrite the `Id` from the body with and id from the query string? If so, could people be relying on this?

Before the entire type `CompositeTestModel` were bound from Body. Eg.

```json
{
 "id" : 1,
 "testModel" : {} 
```
I quite sure someone could be relying on this behavior but, in my opinion, it is wrong and worthy the breaking change.

Also, the user could have the same behavior back adding `FromBody` to the parameter that will ignore all attributes.

What do you think?